src/charm.py

```diff
@@ -199,6 +237,33 @@ def configure_workload_container(self, event) -> None:
                 logger.info(msg)
                 self.unit.status = MaintenanceStatus(msg)
                 container.start(CONTAINER_NAME)
+
+            exporter_container = self.unit.get_container(EXPORTER_CONTAINER_NAME)
+            if exporter_container.can_connect():
+                services = exporter_container.get_plan().to_dict().get("services", {})
+                if services != exporter_config["services"]:
+
+                    msg = "Updating exporter pebble layer config"
+                    logger.info(msg)
+                    self.unit.status = MaintenanceStatus(msg)
+                    exporter_container.add_layer(
+                        EXPORTER_CONTAINER_NAME, exporter_config, combine=True
+                    )
+
+                    service = exporter_container.get_service("nginx-exporter")
+                    if service.is_running():
+                        msg = "Stopping nginx exporter"
+                        logger.info(msg)
+                        self.unit.status = MaintenanceStatus(msg)
+                        exporter_container.stop(EXPORTER_CONTAINER_NAME)
+
+                    msg = "Starting nginx exporter"
+                    logger.info(msg)
+                    self.unit.status = MaintenanceStatus(msg)
+                    exporter_container.start("nginx-exporter")
```

Comment:
You can replan instead to avoid all the logic above: `exporter_container.pebble.replan_services()`